includes/Core/Admin/Screens.php

```diff
@@ -152,14 +152,15 @@ function() {
 		add_filter(
 			'menu_order',
 			function( array $menu_order ) {
-				$new_order = array();
-				foreach ( $menu_order as $index => $item ) {
-					if ( 'index.php' === $item || 0 === strpos( $item, self::PREFIX ) ) {
-						$new_order[] = $item;
-						unset( $menu_order[ $index ] );
-					}
+				// Move the Site Kit dashboard menu item to be one after the index.php item if it exists.
+				$dashboard_index = array_search( 'index.php', $menu_order, true );
+				$sitekit_index   = array_search( self::PREFIX . 'dashboard', $menu_order, true );
```

Comment:
I think we cannot rely on `self::PREFIX . 'dashboard'` here - it was only using `self::PREFIX` before for a reason, which is in certain cases the Site Kit menu is not available under the `dashboard` slug but the `dashboard-splash` slug.

We may need to change this to a more dynamic search which finds the entry that starts with `self::PREFIX` instead.

Reply:
Updated the search to check for self::PREFIX